src/appleseed.shaders/src/appleseed/as_inv_color.osl

```diff
@@ -0,0 +1,159 @@
+// Modeled after Pixar's RenderMan PxrInvert shader.
+
+shader as_inv_color
+[[  
+    string help = "Inverts one or more individual color channels of the the input color in the respective color model. The output is a RGB color.",
+    string icon = "asInvColor.png",
+    string URL = "https://appleseed.readthedocs.io/projects/appleseed-maya/en/latest/shaders/utilities/as_inv_color.html#label-as-inv-color",
+    
+    string as_node_name = "asInvertColor",
+    string as_category = "color",
+    
+    string as_max_class_id = "1105087319 977155459", 
+    string as_max_plugin_type = "texture",
+    
+    int as_maya_type_id = 0x00127a08,
+    string as_maya_classification = "drawdb/shader:rendernode/appleseed/utility"
+]]
+(
+    color in_color = color(0,0,0)
+    [[
+        string as_maya_attribute_name = "color",
+        string as_maya_attribute_short_name = "c",
+        string label = "Input Color",
+        string page = "Color",
+        string help = "Input color.",
+        int divider = 1
+    ]],
+    string in_color_model = "rgb"
+    [[
+        string as_maya_attribute_name = "colormodel",
+        string as_maya_attribute_short_name = "clm",
+        string widget = "popup",
+        string options = "rgb|hsv|hsl|XYZ|xyY",
+        string label = "Color Model",
+        string page = "Color",
+        string help = "The input color uses this color space."
+    ]],
+    int as_invert_channel_0 = 0
+    [[
+        string as_maya_attribute_name = "invert channel 0",
+        string as_maya_attribute_short_name = "inv0",
+        int min = 0,
+        int max = 1,
+        string widget = "checkBox",
+        string label = "Invert Channel 0",
+        string page = "Output",
+        string help = "Inverts the input color channel 0.",
+    ]],
+    int as_invert_channel_1 = 0
+    [[
+        string as_maya_attribute_name = "invert channel 1",
+        string as_maya_attribute_short_name = "inv1",
+        int min = 0,
+        int max = 1,
+        string widget = "checkBox",
+        string label = "Invert Channel 1",
+        string page = "Output",
+        string help = "Inverts the input color channel 1.",
+    ]],
+    int as_invert_channel_2 = 0
+    [[
+        string as_maya_attribute_name = "invert channel 2",
+        string as_maya_attribute_short_name = "inv2",
+        int min = 0,
+        int max = 1,
+        string widget = "checkBox",
+        string label = "Invert Channel 2",
+        string page = "Output",
+        string help = "Inverts the input color channel 2.",
+    ]],
+    output color out_outColor = color(0)
+    [[
+        string as_maya_attribute_name = "outColor",
+        string as_maya_attribute_short_name = "oc",
+        string widget = "null",
+        string label = "Color"
+    ]]
+)
+{
+    color in_color_trans = transformc(in_color_model, in_color);
+    color out_color_trans = color(0);
+    
+    if (as_invert_channel_0 == 1 && as_invert_channel_1 == 0 && as_invert_channel_2 == 0)
```

Comment:
The code can be simplified very significantly:

    out_color_trans = in_color_trans;

    if (as_invert_channel_0 == 1)
         out_color_trans[0] = 1 - in_color_trans[0];

    if (as_invert_channel_1 == 1)
         out_color_trans[1] = 1 - in_color_trans[1];

    if (as_invert_channel_2 == 1)
         out_color_trans[2] = 1 - in_color_trans[2];

Reply:
I tried a simplified logic initially. But after testing I saw it didn't work when selecting 2 or more channels as inverted. So each case is explicitly stated, may look overly complicated but it works reliably. OSL is different from C++ and not all things work equally, e.g. there is no bool data type in OSL.

Reply:
I'm not sure to understand. OSL and C++ are different but still, valid OSL code will behave as expected, and the code I wrote above looks valid to me, so it should do the expected thing.

> But after testing I saw it didn't work when selecting 2 or more channels as inverted.

What happened? Are you sure your code was correct? Did you try mine?

Reply:
You are right. I must have had a bug initially when I tested the simplified version the first time. Your simplification works as expected.